src/imu/imu.cpp

```diff
@@ -0,0 +1,242 @@
+#include <iostream>
+#include <cmath>
+#include <numeric>
+
+#include <robottestingframework/TestAssert.h>
+#include <robottestingframework/dll/Plugin.h>
+
+#include <yarp/os/Property.h>
+#include <yarp/os/Stamp.h>
+
+#include "imu.h"
+
+using namespace robottestingframework;
+ROBOTTESTINGFRAMEWORK_PREPARE_PLUGIN(Imu)
+
+Imu::Imu() : TestCase("Imu") { }
+
+Imu::~Imu() { }
+
+bool Imu::setup(yarp::os::Property& property) {
+    
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("robot"), "The robot name must be given as the test parameter!");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("port"), "The port name must be given as the test parameter!");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("model"), "Please, provide the urdf model path.");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("frame"), "Please, provide the frame name.");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("sensor"), "Please, provide the sensor name.");
+    
+    robotName = property.find("robot").asString(); // robot name
+    portName = property.find("port").asString(); // name of the port from which the data are streamed
+    modelPath = property.find("model").asString(); // urdf model path
+    frameName = property.find("frame").asString(); // frame on which the sensor is attached
+    sensorName = property.find("sensor").asString(); // sensor name within urdf
+
+    ROBOTTESTINGFRAMEWORK_TEST_REPORT("Running IMU test on "+robotName+"...");
+
+    yarp::os::Property options1;
+    options1.put("device", "multipleanalogsensorsclient");
+    options1.put("remote", portName);
+    options1.put("local", "/imuTest/"+portName);
+    
+    driver1 = new yarp::dev::PolyDriver(options1);
+    
+    yarp::os::Property options2;
+    options2.put("device", "remotecontrolboardremapper");
+    yarp::os::Bottle axesNames;
+    yarp::os::Bottle & axesList = axesNames.addList();
+    axesList.addString("neck_pitch");
+    axesList.addString("neck_roll");
+    axesList.addString("neck_yaw");
+    axesList.addString("torso_pitch");
+    axesList.addString("torso_roll");
+    axesList.addString("torso_yaw");
+    axesList.addString("l_shoulder_pitch");
+    axesList.addString("l_shoulder_roll");
+    axesList.addString("l_shoulder_yaw");
+    axesList.addString("l_elbow");
+    axesList.addString("r_shoulder_pitch");
+    axesList.addString("r_shoulder_roll");
+    axesList.addString("r_shoulder_yaw");
+    axesList.addString("r_elbow");
```

Comment:
This list can be parametrized in future improvements in order to be able to test sensors in other kinematic chains